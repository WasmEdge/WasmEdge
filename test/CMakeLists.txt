# SPDX-License-Identifier: Apache-2.0

if (NOT SSVM_DISABLE_AOT_RUNTIME)
  add_subdirectory(aot)
endif()
add_subdirectory(spec)
add_subdirectory(ast)
add_subdirectory(loader)
add_subdirectory(interpreter)
add_subdirectory(expected)
add_subdirectory(span)

if(BUILD_COVERAGE)
  setup_target_for_coverage_gcovr_html(
    NAME coverage
    EXECUTABLE ctest -j ${PROCESSOR_COUNT}
    DEPENDENCIES ssvm
    BASE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
    EXCLUDE
      "${PROJECT_SOURCE_DIR}/thirdparty/*"
      "${PROJECT_SOURCE_DIR}/test/*"
    )
  setup_target_for_coverage_gcovr_xml(
    NAME codecov
    EXECUTABLE ctest -j ${PROCESSOR_COUNT}
    DEPENDENCIES ssvm
    BASE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
    EXCLUDE
      "${PROJECT_SOURCE_DIR}/thirdparty/*"
      "${PROJECT_SOURCE_DIR}/test/*"
    )

endif()
