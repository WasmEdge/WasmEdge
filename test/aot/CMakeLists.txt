# SPDX-License-Identifier: Apache-2.0

add_executable(ssvmAOTWagonTests
  AOTwagonTest.cpp
)

add_test(ssvmAOTWagonTests ssvmAOTWagonTests)

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/../loader/wagonTestData
  ${CMAKE_CURRENT_BINARY_DIR}/../loader/wagonTestData
  COPYONLY
)

target_link_libraries(ssvmAOTWagonTests
  PRIVATE
  utilGoogleTest
  ssvmLoader
  ssvmAOT
)

add_executable(ssvmAOTCoreTests
  AOTcoreTest.cpp
)

add_test(ssvmAOTCoreTests ssvmAOTCoreTests)

target_link_libraries(ssvmAOTCoreTests
  PRIVATE
  std::filesystem
  utilGoogleTest
  ssvmTestSpec
  ssvmLoader
  ssvmAOT
  ssvmVM
)
