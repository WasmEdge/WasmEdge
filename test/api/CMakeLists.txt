# SPDX-License-Identifier: Apache-2.0
# SPDX-FileCopyrightText: 2019-2022 Second State INC

if(WASMEDGE_BUILD_AOT_RUNTIME)
  add_definitions(-DWASMEDGE_BUILD_AOT_RUNTIME)
endif()
if(WASMEDGE_BUILD_PLUGINS)
  add_definitions(-DWASMEDGE_BUILD_PLUGINS)
endif()

file(COPY
  ${CMAKE_CURRENT_SOURCE_DIR}/apiTestData
  DESTINATION
  ${CMAKE_CURRENT_BINARY_DIR}
)

wasmedge_add_library(wasmedgeAPITestHelpers
  hostfunc_c.c
  helper.cpp
)

target_link_libraries(wasmedgeAPITestHelpers
  PRIVATE
  wasmedge_shared
  wasmedgeTestSpec
)

wasmedge_add_executable(wasmedgeAPIUnitTests
  APIUnitTest.cpp
)

add_test(wasmedgeAPIUnitTests wasmedgeAPIUnitTests)

target_link_libraries(wasmedgeAPIUnitTests
  PRIVATE
  std::filesystem
  Boost::boost
  ${GTEST_BOTH_LIBRARIES}
  wasmedgeAPITestHelpers
  wasmedge_shared
)

target_include_directories(wasmedgeAPIUnitTests
  PRIVATE
  ${CMAKE_SOURCE_DIR}/include
)

wasmedge_add_executable(wasmedgeAPIVMCoreTests
  APIVMCoreTest.cpp
)

add_test(wasmedgeAPIVMCoreTests wasmedgeAPIVMCoreTests)

target_link_libraries(wasmedgeAPIVMCoreTests
  PRIVATE
  std::filesystem
  ${GTEST_BOTH_LIBRARIES}
  wasmedgeTestSpec
  wasmedgeAPITestHelpers
  wasmedge_shared
)

wasmedge_add_executable(wasmedgeAPIStepsCoreTests
  APIStepsCoreTest.cpp
)

add_test(wasmedgeAPIStepsCoreTests wasmedgeAPIStepsCoreTests)

target_link_libraries(wasmedgeAPIStepsCoreTests
  PRIVATE
  std::filesystem
  ${GTEST_BOTH_LIBRARIES}
  wasmedgeTestSpec
  wasmedgeAPITestHelpers
  wasmedge_shared
)

if(WASMEDGE_BUILD_AOT_RUNTIME)
  wasmedge_add_executable(wasmedgeAPIAOTCoreTests
    APIAOTCoreTest.cpp
  )

  add_test(wasmedgeAPIAOTCoreTests wasmedgeAPIAOTCoreTests)

  target_link_libraries(wasmedgeAPIAOTCoreTests
    PRIVATE
    std::filesystem
    ${GTEST_BOTH_LIBRARIES}
    wasmedgeTestSpec
    wasmedgeAPITestHelpers
    wasmedge_shared
  )
endif()
