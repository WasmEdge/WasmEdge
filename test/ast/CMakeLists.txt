# SPDX-License-Identifier: Apache-2.0

wasmedge_add_executable(wasmedgeASTTests
  moduleTest.cpp
  sectionTest.cpp
  descriptionTest.cpp
  segmentTest.cpp
  typeTest.cpp
  expressionTest.cpp
  instructionTest.cpp
)

add_test(wasmedgeASTTests wasmedgeASTTests)

target_link_libraries(wasmedgeASTTests
  PRIVATE
  ${GTEST_BOTH_LIBRARIES}
  wasmedgeLoaderFileMgr
  wasmedgeAST
)
