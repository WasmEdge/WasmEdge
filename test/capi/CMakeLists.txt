# SPDX-License-Identifier: Apache-2.0
# SPDX-FileCopyrightText: 2019-2024 Second State INC

wasmedge_add_executable(capiTests
  spectest.cpp
)

add_test(capiTests capiTests)

target_link_libraries(capiTests
  PRIVATE
  wasmedgeCAPI
  ${GTEST_BOTH_LIBRARIES}
)

target_include_directories(capiTests
  PRIVATE
  ${PROJECT_SOURCE_DIR}/include
)
