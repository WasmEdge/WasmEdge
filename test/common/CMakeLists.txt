# SPDX-License-Identifier: Apache-2.0
# SPDX-FileCopyrightText: 2019-2022 Second State INC

wasmedge_add_executable(wasmedgeCommonTests
  int128Test.cpp
)

add_test(wasmedgeCommonTests wasmedgeCommonTests)

target_link_libraries(wasmedgeCommonTests
  PRIVATE
  ${GTEST_BOTH_LIBRARIES}
  wasmedgeCommon
)
