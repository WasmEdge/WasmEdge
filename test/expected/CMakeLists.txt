# SPDX-License-Identifier: Apache-2.0
# SPDX-FileCopyrightText: 2019-2022 Second State INC

wasmedge_add_executable(expectedTests
  assignment.cpp
  bases.cpp
  constexpr.cpp
  constructors.cpp
  emplace.cpp
  extensions.cpp
  gtest.cpp
  issues.cpp
  noexcept.cpp
  observers.cpp
  relops.cpp
  swap.cpp
)

add_test(expectedTests expectedTests)

target_link_libraries(expectedTests
  PRIVATE
  ${GTEST_BOTH_LIBRARIES}
)

target_include_directories(expectedTests
  PRIVATE
  ${PROJECT_SOURCE_DIR}/include
)
